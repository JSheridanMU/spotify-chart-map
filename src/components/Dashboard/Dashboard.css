.VolumeSlider {
    z-index: 1000;
    position: absolute;
    width: 150px;
    right: 20px;
    top: 10px;
}

table tbody, table thead {
    display: block;
}

.table td {
    border-color: #201f1f !important;
}

.table th {
    border-color: #201f1f !important;
}

.table td {
    height: 75px;
    vertical-align: middle !important;
}

.table-hover tbody tr:hover td {
    background:#201f1f;
    color: white;
}

table tbody {
    max-height: 480px;
    overflow-y: scroll;
    color: white;
    background: #181717;
}

thead > tr > th {
    background: #181717;
    color: white;
}

th:nth-child(1), td:nth-child(1) {
    width: 75px;
}

th:nth-child(2), td:nth-child(2) {
    width: 200px;
}

th:nth-child(3) {
    width: 100px;
}