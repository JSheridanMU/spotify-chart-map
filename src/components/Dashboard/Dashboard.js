import React, { useState } from "react";
import { Col, Container, Row, Table } from "react-bootstrap";
import RangeSlider from "react-bootstrap-range-slider";
import MapView from "../MapView/MapView";
import GetCountry from "../GetCountry/GetCountry";
//import GetCharts from "../GetCharts/GetCharts"
//import ChartsData from "../ChartsData/ChartsData"
import "./Dashboard.css";
import moment from "moment";
import Header from "../Header/Header";
import Player from "../Player/Player";

const initialCountry = {
  name: "Ireland",
  code: "ie",
};

export default function Dashboard() {
  let dateArray = [];
  let minDate = "01/01/2016";
  let maxDate = moment().subtract(1, "days").format("DD/MM/YYYY"); //may need to change to subtract 2 days depending on when spotify updates

  while (
    moment(minDate, "DD/MM/YYYY").valueOf() <= moment(maxDate, "DD/MM/YYYY").valueOf()
  ) {
    dateArray.push(moment(minDate, "DD/MM/YYYY").format("DD/MM/YYYY"));
    minDate = moment(minDate, "DD/MM/YYYY").add(1, "days").format("DD/MM/YYYY");
  }

  const { country, handleInputChange } = GetCountry(initialCountry);
  const [tempDate, setTempDate] = useState(maxDate);
  const [finalDate, setFinalDate] = useState(tempDate);

  const [_token, setToken] = useState(null);

  const handleTokenChange = (e) => {
    setToken(e)
  }
  
  return (
    <React.Fragment>
      <Header tokenChange={handleTokenChange}/>
      <Container>
        <Row className="justify-content-md-center">
          <Col xs={6}>
            <RangeSlider
              value={tempDate}
              variant={"success"}
              onChange={(changeEvent) => setTempDate(changeEvent.target.value)}
              onAfterChange={(changeEvent) =>
                setFinalDate(changeEvent.target.value)
              }
              min={0}
              max={dateArray.length - 1}
              tooltip={"auto"}
              tooltipLabel={(currentValue) => dateArray[tempDate]}
            />
          </Col>
        </Row>
        <Row>
          <Col xs={3}>
            <h3 style={{ color: "#1DB954" }}>
              Global Charts{" "}
              {dateArray[finalDate] ? dateArray[finalDate] : maxDate}
            </h3>
            <Table className="GlobalChartTable"></Table>
          </Col>
          <Col xs={6}>
            <MapView handleInputChange={handleInputChange} />
          </Col>
          <Col xs={3}>
            <h3 style={{ color: "#1DB954" }}>
              {country.name} Charts{" "}
              {dateArray[finalDate] ? dateArray[finalDate] : maxDate}
            </h3>
            <Table className="CountryChartTable"></Table>
          </Col>
        </Row>
        <Row>
          {_token && (
              <Player
                token={_token}
              />
            )}
        </Row>
      </Container>
    </React.Fragment>
  );
}
