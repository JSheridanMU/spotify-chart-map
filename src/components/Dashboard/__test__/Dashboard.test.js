import * as React from "react";
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Dashboard from '.././Dashboard';

afterEach(cleanup)

test('renders without crashing', () => {
  render(<Dashboard/>);
});

test('detects page content', () => {
  const {getByText, getAllByText, getByTestId} = render(<Dashboard/>);

  //table titles
  getByText('Global Charts 11/12/2020')
  getByText('Ireland Charts 11/12/2020')

  //table headers
  getAllByText('Rank')
  getAllByText('Song')
  getAllByText('Streams')

  //slider
  getByTestId('slider')
  
  //table data
  getAllByText('Loading')
});

test('matches snapshot', () => {
    const tree = renderer.create(<Dashboard/>).toJSON()
    expect(tree).toMatchSnapshot()
});