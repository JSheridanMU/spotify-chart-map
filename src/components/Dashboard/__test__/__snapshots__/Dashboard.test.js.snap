// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <nav
    className="color-nav navbar navbar-expand navbar-dark"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      <img
        alt=""
        className="d-inline-block align-top"
        height="30"
        src="/logo.png"
        width="30"
      />
       
      Spotify Chart Map
    </a>
    <a
      className="btn btn--loginApp-link ml-auto"
      href="https://accounts.spotify.com/authorize?client_id=d6aa49c0c90c42a49e4aa1e2b04dfb63&redirect_uri=http://localhost:3000&scope=streaming%20user-read-email%20user-read-private%20user-read-playback-state%20user-library-read%20user-library-modify%20user-read-currently-playing%20user-modify-playback-state&response_type=token&show_dialog=true"
    >
      Login to Spotify
    </a>
  </nav>,
  <div
    className="container-fluid"
  >
    <div
      className="justify-content-md-center row"
    >
      <div
        className="col-6"
      >
        <div
          data-testid="slider"
        >
          <span
            className="range-slider__wrap"
          >
            <input
              className="range-slider range-slider--success"
              disabled={false}
              max={1440}
              min={0}
              onChange={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              type="range"
              value={1440}
            />
            <div
              className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
              style={
                Object {
                  "left": "calc(100% + -10px)",
                }
              }
            >
              <div
                className="range-slider__tooltip__label"
              >
                11/12/2020
              </div>
              <div
                className="range-slider__tooltip__arrow"
              />
            </div>
          </span>
        </div>
      </div>
    </div>
    <div
      className="row no-gutters"
    >
      <div
        className="col-3"
      >
        <h4
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          Global Charts
           
          11/12/2020
        </h4>
        <div
          className="GlobalChartTable"
        >
          <table
            className="table table-hover"
          >
            <thead>
              <tr>
                <th>
                  Rank
                </th>
                <th>
                  Song
                </th>
                <th>
                  Streams
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  !!!
                </td>
                <td>
                  Loading
                </td>
                <td>
                  !!!
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        className="col-6"
      >
        <div
          className="map-container container"
        >
          <div
            style={
              Object {
                "height": "70vh",
                "width": "35vw",
              }
            }
          />
        </div>
      </div>
      <div
        className="col-3"
      >
        <h4
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          Ireland
           Charts
           
          11/12/2020
        </h4>
        <div
          className="CountryChartTable"
        >
          <table
            className="table table-hover"
          >
            <thead>
              <tr>
                <th>
                  Position
                </th>
                <th>
                  Track
                </th>
                <th>
                  Streams
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  !!!
                </td>
                <td>
                  Loading
                </td>
                <td>
                  !!!
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      className="row"
    />
  </div>,
]
`;
